packages/driver/cypress/integration/e2e/multi-domain/multi_domain_uncaught_errors_spec.ts

```diff
@@ -240,21 +257,220 @@ describe('multi-domain - uncaught errors', { experimentalSessionSupport: true },
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw document.createElement('h1')
+      })
+    })
+
+    it('handles users throwing functions', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw () => undefined
+      })
+    })
+
+    it('handles users throwing symbols', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw Symbol('foo')
+      })
+    })
+
+    it('handles users throwing promises', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
 
-      done()
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw new Promise(() => {})
+      })
     })
+  })
+
+  describe('serializable errors', () => {
+    it('handles users throwing complex errors/classes', (done) => {
+      cy.on('fail', (err: any) => {
+        expect(err.name).to.equal('CustomError')
+        expect(err.message).to.equal('custom error')
+        expect(err._name).to.equal('CustomError')
+        expect(err.foo).to.equal('bar')
 
-    cy.switchToDomain('foobar.com', () => {
-      cy.then(() => {
-        throw new class CustomError extends Error {
+        const { writable } = Object.getOwnPropertyDescriptor(err, 'name') || {}
+
+        // After serialization, read-only properties are now writable
+        expect(writable).to.be.true
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        class CustomError extends Error {
+          private _name = 'CustomError'
           get name () {
-            return 'CustomError'
+            return this._name
+          }
+          set name (name: string) {
+            this._name = name
+          }
+        }
+
+        const customErrorInstance = new CustomError('custom error')
+
+        // @ts-ignore
+        customErrorInstance.foo = 'bar'
+
+        const { writable } = Object.getOwnPropertyDescriptor(CustomError, 'name') || {}
```

Comment:
```suggestion
        const { writable } = Object.getOwnPropertyDescriptor(CustomError, 'name') as PropertyDescriptor
```

Reply:
Since this is the test code and we have absolute control of `CustomError` I am good with this. Will update!

Reply:
addressed in 4872216566f6ad7b26933bb99def80295f04e236